daemon/docker_watcher.go

```diff
@@ -238,13 +238,8 @@ func (d *Daemon) handleCreateContainer(id string, retry bool) {
 			if cid != 0 {
 				endpointmanager.Mutex.Lock()
 				ep = endpointmanager.LookupCiliumIDLocked(cid)
-				if ep != nil {
-					// Associate container id with endpoint
-					ep.Mutex.Lock()
-					ep.DockerID = id
-					ep.Mutex.Unlock()
-					endpointmanager.LinkContainerID(ep)
-				}
+				// Associate container id with endpoint
+				endpointmanager.LinkContainerID(ep)
```

Comment:
Calling `LinkContainerID` only makes sense after setting `ep.DockerID`, in this branch, we know that the endpoint exists (it was retrievable by endpoint ID but not by DockerID). I think we should keep the code to set `ep.DockerID`